qutebrowser/misc/sessions.py

```diff
@@ -347,6 +354,19 @@ def _load_tab(self, new_tab, data):
             if 'pinned' in histentry:
                 new_tab.data.pinned = histentry['pinned']
 
+            if (config.val.session_lazy_restore and
+                    histentry.get('active', False) and
+                    not histentry['url'].startswith('qute://back')):
+                # remove "active" mark and insert back page marked as active
+                data['history'].insert(
```

Comment:
Here you're mutating the list you're iterating over. I'm not sure whether this is guaranteed to work.

Reply:
Well, it's recommened not to change a list during iteration, but in this case it seems to work.

> If you need to modify the sequence you are iterating over while inside the loop (for example to duplicate selected items), it is recommended that you first make a copy. Iterating over a sequence does not implicitly make a copy. 

Source: https://docs.python.org/3/tutorial/controlflow.html#for-statements
Test:

```python
>>> l=[i for i in range(1, 10)]
>>> for i, n in enumerate(l):
        if n == 5:
            l.insert(i+1, 999)
        print(i, ':', n)
0 : 1
1 : 2
2 : 3
3 : 4
4 : 5
5 : 999
6 : 6
7 : 7
8 : 8
9 : 9
```

However I could rewrite it to e.g.:
```python
        # ...
        lazy_load = []
        # use len(data['history']) -> dropwhile empty if not session.lazy_session
        lazy_index = len(data['history'])
        gen = chain(
                takewhile(lambda *_: not lazy_load, enumerate(data['history'])),
                enumerate(lazy_load),
                dropwhile(lambda i: i[0] < lazy_index, enumerate(data['history'])))

        for i, histentry in gen:
            # ...
            if (config.val.session.lazy_restore and
                    histentry.get('active', False) and
                    not histentry['url'].startswith('qute://back')):
                # remove "active" mark and insert back page marked as active
                lazy_index = i + 1
                lazy_load.append({
                        'title': histentry['title'],
                        'url': 'qute://back#' + histentry['title'],
                        'active': True
                    })
                histentry['active'] = False
            # ...
```

As chain is defined as:
```python
def chain(*iterables):
    for it in iterables: # [takewhile, lazy_load, slice]
        for element in it: 
            yield element
```
We would modify lazy_load while iterating over iterables and takewhile  
-> same as modifing an element of a list.